mm2src/coins/eth.rs

```diff
@@ -796,6 +811,200 @@ async fn withdraw_impl(coin: EthCoin, req: WithdrawRequest) -> WithdrawResult {
+                if !response["cursor"].is_null() {
+                    if let Some(cursor_res) = response["cursor"].as_str() {
+                        cursor = format!("{}{}", "&cursor=", cursor_res);
+                        continue;
+                    }
+                } else {
+                    break;
+                }
+            }
+        }
+    }
+    drop_mutability!(res_list);
+    let nft_list = NftList {
+        count: res_list.len() as u64,
+        nfts: res_list,
+    };
+    Ok(nft_list)
+}
+
+/// `get_nft_metadata` function returns info of one specific NFT.
+/// Current implementation sends request to Moralis.
+/// Later, after adding caching, metadata lookup can be performed using previously obtained NFTs info without
+/// sending new moralis request. The moralis request can be sent as a fallback, if the data was not found in the cache.
+pub async fn get_nft_metadata(ctx: MmArc, req: NftMetadataReq) -> MmResult<Nft, GetNftInfoError> {
+    let api_key = match ctx.conf["api_key"].as_str() {
+        Some(api_key) => api_key,
+        None => return Err(MmError::new(GetNftInfoError::ApiKeyError)),
+    };
+    let chain_str = match req.chain {
+        Chain::Bnb => "bsc",
+        Chain::Eth => "eth",
+    };
+    let uri = format!(
+        "{}nft/{}/{}?chain={}&{}",
+        URL_MORALIS, req.token_address, req.token_id, chain_str, FORMAT_DECIMAL_MORALIS
+    );
+    let response = send_moralis_request(uri.as_str(), api_key).await?;
+    let nft_wrapper: NftWrapper = serde_json::from_str(&response.to_string())?;
+    let nft_metadata = Nft {
+        chain: req.chain,
+        token_address: nft_wrapper.token_address,
+        token_id: nft_wrapper.token_id.0,
+        amount: nft_wrapper.amount.0,
+        owner_of: nft_wrapper.owner_of,
+        token_hash: nft_wrapper.token_hash,
+        block_number_minted: *nft_wrapper.block_number_minted,
+        block_number: *nft_wrapper.block_number,
+        contract_type: nft_wrapper.contract_type.map(|v| v.0),
+        name: nft_wrapper.name,
+        symbol: nft_wrapper.symbol,
+        token_uri: nft_wrapper.token_uri,
+        metadata: nft_wrapper.metadata,
+        last_token_uri_sync: nft_wrapper.last_token_uri_sync,
+        last_metadata_sync: nft_wrapper.last_metadata_sync,
+        minter_address: nft_wrapper.minter_address,
+    };
+    Ok(nft_metadata)
+}
+
+/// `get_nft_transfers` function returns a transfer history of NFTs on ETH or/and BNb chains owned by user.
+/// Currently doesnt support filters.
+pub async fn get_nft_transfers(ctx: MmArc, req: NftTransfersReq) -> MmResult<NftsTransferHistoryList, GetNftInfoError> {
+    let api_key = match ctx.conf["api_key"].as_str() {
+        Some(api_key) => api_key,
+        None => return Err(MmError::new(GetNftInfoError::ApiKeyError)),
+    };
+
+    let mut res_list = Vec::new();
+
+    for chain in req.chains {
+        let (coin_str, chain_str) = match chain {
+            Chain::Bnb => ("BNB", "bsc"),
```

Comment:
BNB is the platform coin, but the chain is called BSC. How about renaming `Chain::Bnb` to `Chain::Bsc`?

Reply:
As far as I know it was rebranded to [BNB chain](https://cointelegraph.com/news/binance-smart-chain-becomes-bnb-chain). BSC, crypto exchange Binance’s layer-1 blockchain, recently announced the merger of the Binance Smart Chain and the Binance Chain [to form the BNB Chain](https://cointelegraph.com/altcoins-for-beginners/a-beginners-guide-to-the-bnb-chain-the-evolution-of-the-binance-smart-chain). Also currently in design examples we display chains as: `BNB blockchain` and `ETH blockchain`.
Actually, I was confused too, but decided to leave chain as BNB.

Reply:
Sorry, now I think you was right that Bsc is better,
I wanted to test NFT withdrawing and found out that different exchanges/wallets can call it `BNB Chain` or `BNB Smart Chain`.
Interesting example [has OpenSea](https://opensea.io/assets/bsc/0x3d8ce31601b8ed4db5eeb921fcf7f178babe7c2e/1744).
In details description we see a chain type as BNB chain, while in url they use bsc.
May be its not so sensitive when user uses gui, however for API side it can cause misunderstanding. Changed `Chain::Bnb` to `Chain::Bsc`.

Reply:
Great, thank you!